File: submissions/tngu281.cs
using System;
using System.IO;
using System.Linq;
using System.Text.RegularExpressions;

class Program
{
    static void Main(string[] args)
    {
        int k = 3;
        try
        {
            if (args.Length == 2)
            {
                if (!Regex.IsMatch(args[1], @"^\+?([0-9]\d*)$")) throw new Exception($"{args[1]} is not a valid number");
                k = int.Parse(args[1]);
            }
            else if (args.Length != 1)
            {
                throw new Exception("wrong number of arguments");
            }
            Regex
                .Matches(File.ReadAllText(args[0]).ToUpper(), @"[A-Z]+")
                .Cast<Match>()
                .GroupBy(word => word.Value)
                .OrderByDescending(word => word.Count())
                .ThenBy(word => word.Key)
                .Take(k)
                .ToList()
                .ForEach(word => Console.WriteLine($"{word.Key} {word.Count()}"));
        }
        catch (Exception ex)
        {
            Console.WriteLine($"*** Error: {ex.Message}");
            Environment.ExitCode = 1;
        }
    }
}

File: submissions/tngu281.fs
open System
open System.IO
open System.Text.RegularExpressions

[<EntryPoint>]
let main argv =
    try
        if argv.Length <> 2 then failwith "wrong number of arguments"
        if not (Regex.IsMatch(argv.[1], @"^\+?(0|[1-9]\d*)$")) then failwith "k is not a valid number"
        if not (Regex.IsMatch(argv.[0], @".txt$")) then failwith (argv.[0] + " is not a valid txt file") 
        File.ReadAllText(argv.[0]).ToUpper()
        |> (fun(x:String)-> Regex.Matches(x,@"[A-Z]+"))
        |> Seq.cast
        |> Seq.map(fun(x:Match)->x.Value)
        |> Seq.countBy id
        |> Seq.sortBy(fun(x,y)->(-y,x))
        |> Seq.take(int(argv.[1]))
        |> Seq.iter(fun(x:String,y:int)->printfn "%s %i" x y)
    with
        |  ex -> printfn "*** Error: %s" ex.Message
    0 // return an integer exit code